AVADA.pm

```diff
@@ -0,0 +1,172 @@
+monogenic disease and convert it to genomic coordinates.
+
+Please cite the AVADA publication alongside the VEP if you use this resource:
+https://pubmed.ncbi.nlm.nih.gov/31467448/
+
+Pre-requisites
+AVADA data can be downloaded from: 
+http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz
+
+wget http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz
+gzip -d avada_v1.00_2016.vcf.gz
+bgzip -c avada_v1.00_2016.vcf > avada_v1.00_2016.vcf.gz
+tabix -p vcf avada_v1.00_2016.vcf.gz
+Output: 
+The output includes two columns  AVADA_FEATURE_ID; AVADA_PMID
+
+=cut
+
+package AVADA;
+
+use strict;
+use warnings;
+use Data::Dumper;
+use List::MoreUtils qw(uniq);
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+
+
+my $feature_match_by;
+
+sub get_header_info {
+  my $self = shift;
+  my %header;
+
+  $header{"AVADA_PMID"} = "PubMed ID evidence for the variant as reported by AVADA";
+  $header{"AVADA_FEATURE_ID"} = "Feature ID associated with variant as reported by AVADA" ;  
+  return \%header
+}
+
+sub new {
+  my $class = shift;
+  
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+
+  $self->get_user_params();
+  my $param_hash = $self->params_to_hash();
+  my $file = $param_hash->{file};
+  die "\n  ERROR: No file specified\nTry using 'file=path/to/file.tsv.gz'\n" unless defined($file);
+  $self->add_file($file);
+  $feature_match_by = $param_hash->{feature_match_by}; 
+  if (defined($feature_match_by) && $feature_match_by  eq "refseq_transcript_id") {
+    $self->{config}->{refseq} = 1 unless $self->{config}->{database} == 1 ; # when using db, refseq is not enabled in spite of forcing in code
+    # Currently sets use_given_ref to 1 always
+    $self->{config}->{use_given_ref} = 1;
+    die "\n ERROR: Matching by Refseq ID requires the option --refseq when using the database  \n" if $self->{config}->{refseq} == 0 && $self->{config}->{database} == 1;
+  }
+  return $self;
+}
+
+sub run {
+  my ($self, $tva) = @_;
+  my $vf = $tva->variation_feature;
+  my $tv = $tva->transcript_variation;
+  my $end = $vf->{end};
+  my $start = $vf->{start};
+  my $transcript = $tva->transcript;
+  my %output;
+  my @data;
+
+  if (!defined($feature_match_by)){
+    @data = @{$self->get_data($vf->{chr}, $start, $end)};
+    $output{"AVADA_FEATURE_ID"} = $data[0]->{AVADA_GENE_SYMBOL}  if scalar @data && defined $data[0]->{AVADA_GENE_SYMBOL} ; 
+  }
+  elsif ( $feature_match_by eq "gene_symbol"){
+    @data = grep {
+    $_->{AVADA_GENE_SYMBOL} eq $transcript->{_gene_symbol}
+    }@{$self->get_data($vf->{chr}, $start, $end)};
+    $output{"AVADA_FEATURE_ID"} = $data[0]->{AVADA_GENE_SYMBOL} if scalar @data && defined $data[0]->{AVADA_GENE_SYMBOL} ; ;
+  }
+  elsif ( $feature_match_by eq "refseq_transcript_id" ){
+    my $refseq_transcript = $transcript->{stable_id};
```

Comment:
sometimes only the RefSeq protein id is listed for a paper/variant and not the RefSeq transcript id, so this would lose data